gtsfm/bundle/bundle_adjustment.py

```diff
@@ -28,6 +41,78 @@ class BundleAdjustmentOptimizer:
     This class refines global pose estimates and intrinsics of cameras, and also refines 3D point cloud structure given
     tracks from triangulation."""
 
+    def __init__(self, shared_calib: bool = False):
+        """Initializes the parameters for bundle adjustment module.
+
+        Args:
+            shared_calib (optional): Flag to enable shared calibration across
+                                     all cameras. Defaults to False.
+        """
+
+        self._shared_calib = shared_calib
+
+    def __add_camera_prior_and_initial_value(
+        self, graph: NonlinearFactorGraph, initial_values: Values, camera: PinholeCameraCal3Bundler, camera_idx: int,
+    ) -> None:
+        """Add a prior factor in the factor graph and initial values for the
+        camera parameters.
+
+        Args:
+            graph: factor graph for the problem.
+            initial_values: object holding the initial values for camera
+                            parameters and 3d points.
+            camera: the camera object to add to graph and to initial values.
+            camera_idx: index of the camera.
+        """
+
+        # add prior factor for pose
+        graph.push_back(PriorFactorPose3(X(camera_idx), camera.pose(), Isotropic.Sigma(CAM_POSE3_DOF, 0.1),))
```

Comment:
why are we putting a prior factor on every pose3?

before it was just on the first pose

Reply:
I remember Frank mentioning that we should use priors on camera poses because we have good initializations from the averaging module. Maybe this needs to be tested first in a separate PR?